pybandits/base_model.py

```
@@ -215,11 +226,16 @@ class BaseModelMO(BaseModel, ABC):
-    def sample_proba(self, **kwargs) -> Union[List[MOProbability], List[QuantitativeMOProbability]]:
+    def sample_proba(self, rng: Generator, **kwargs) -> Union[List[MOProbability], List[QuantitativeMOProbability]]:
         """
         Sample the probability of getting a positive reward.
+
+        Parameters
+        ----------
+        rng : numpy.random.Generator
+            Central numpy random generator provided by the MAB.
         """
-        return [list(p) for p in zip(*[model.sample_proba(**kwargs) for model in self.models])]
+        return [list(p) for p in zip(*[model.sample_proba(rng=rng, **kwargs) for model in self.models])]
```

Ruff (0.15.9)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)